core/types/block_ext_test.go

```diff
@@ -0,0 +1,105 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package types
+
+import (
+	"math/big"
+	"reflect"
+	"testing"
+	"unsafe"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCopyHeader(t *testing.T) {
+	t.Parallel()
+
+	t.Run("empty_header", func(t *testing.T) {
+		t.Parallel()
+
+		empty := &Header{}
+
+		headerExtra := &HeaderExtra{}
+		extras.Header.Set(empty, headerExtra)
+
+		cpy := CopyHeader(empty)
+
+		want := &Header{
+			Difficulty: new(big.Int),
+			Number:     new(big.Int),
+		}
+
+		headerExtra = &HeaderExtra{}
+		extras.Header.Set(want, headerExtra)
+
+		assert.Equal(t, want, cpy)
+	})
+
+	t.Run("filled_header", func(t *testing.T) {
+		t.Parallel()
+
+		header, _ := headerWithNonZeroFields() // the header carries the [HeaderExtra] so we can ignore it
+
+		gotHeader := CopyHeader(header)
+		gotExtra := GetHeaderExtra(gotHeader)
+
+		wantHeader, wantExtra := headerWithNonZeroFields()
+		assert.Equal(t, wantHeader, gotHeader)
+		assert.Equal(t, wantExtra, gotExtra)
+
+		exportedFieldsDeepCopied(t, header, gotHeader)
+		exportedFieldsDeepCopied(t, GetHeaderExtra(header), gotExtra)
+	})
+}
+
+func exportedFieldsDeepCopied[T interface {
```

Comment:
should we rename this? `exportedFieldsPointToDifferentMemory`?

Reply:
Sure 😄 That was the original name written by ARR4N (well `fieldsPointToDifferentMemory`) and thought it was too many syllables - changed to `exportedFieldsPointToDifferentMemory`